### PR TITLE
Introduction of Arrays

### PR DESCRIPTION
**Pull request by danielwilms**
* Added definition of arrays
* Fixed minor issues in branches

Fixes: #110

### CODE REVIEW

**Comment by danielwilms** [comment]
@danielwilms: add DTC example

**Comment by danielwilms** [comment]
> I guess it shall be applicable to all data types except string? If so, maybe that should be mentioned.

I'd include strings there. Why would you exclude them @UlfBj?

**Comment by UlfBj** [comment]
Because a string is by itself an unlimited array.
So I thought it could become complicated with an unlimited array of
unlimited array.
I am ok to include it.
BR
Ulf

**Comment by danielwilms** [comment]
> Because a string is by itself an unlimited array. So I thought it could become complicated with an unlimited array of unlimited array. I am ok to include it.

I would see the danger, that the array is built then within the string, which then leads to parsing on the client side. That made me think, it might be better to make it explicit.

**Comment by danielwilms** [comment]
* best practice documentation

**Comment by gunnarx** [comment]
I was sure I had saved my review last week, but apparently it was not there.  I just went in and pressed save now, so it should be visible.

File: docs-gen/content/rule_set/data_entry/data_types.md [hunk 1]
```diff
@@ -22,4 +24,20 @@ Double     | double precision floating point number | -1.7e -300 | 1.7e 300
...
 ByteBuffer | buffer of bytes (aka BLOB) | n/a | n/a
 
-Datatypes are not specified for ```branch```
+
+## Arrays
+
+Besides the datatypes described above, VSS supports as well the concept of
+`arrays`, as a *collection of elements based on the data entry
+definition*, wherein it's specified. Both - limited and
+unlimited - types of arrays are supported. The following syntax shall be used to declare an array:
```
[the remark this answers is not in the document]
Comment (SebastianSchildt) [reply docs-gen/content/rule_set/data_entry/data_types.md:33]: fixed or variable size arrays

**Pull request merged**